src/test/java/ReceiptAnalysisTest.java

```diff
@@ -275,6 +276,70 @@ public void analyzeImageAt_dateAndPrice_returnsTimestampAndPrice()
     Assert.assertEquals(PRICE, results.getPrice());
   }
 
+  @Test
+  public void analyzeImageAt_priceAfterTotal_returnsPrice()
+      throws IOException, ReceiptAnalysisException {
+    String rawTextWithPriceTotal = "the total is $" + PRICE_VALUE;
+    stubAnnotationResponse(LOGO_CONFIDENCE, rawTextWithPriceTotal);
+    stubTextClassification();
+
+    AnalysisResults results = ReceiptAnalysis.analyzeImageAt(url);
+
+    Assert.assertEquals(PRICE, results.getPrice());
+  }
+
+  @Test
+  public void analyzeImageAt_multipleTotals_returnsPriceAfterLastTotal()
+      throws IOException, ReceiptAnalysisException {
+    String rawTextWithPriceMultipleTotals =
+        "the sub total is $12.23 and the total is $" + PRICE_VALUE;
+    stubAnnotationResponse(LOGO_CONFIDENCE, rawTextWithPriceMultipleTotals);
+    stubTextClassification();
+
+    AnalysisResults results = ReceiptAnalysis.analyzeImageAt(url);
+
+    Assert.assertEquals(PRICE, results.getPrice());
+  }
+
+  @Test
+  public void analyzeImageAt_lastTotalHasNoPrice_returnsPriceAfterOtherTotal()
+      throws IOException, ReceiptAnalysisException {
+    String rawTextWithPriceExtraTotal =
+        "the total is $" + PRICE_VALUE + " which is the total that must be paid";
+    stubAnnotationResponse(LOGO_CONFIDENCE, rawTextWithPriceExtraTotal);
+    stubTextClassification();
+
+    AnalysisResults results = ReceiptAnalysis.analyzeImageAt(url);
+
+    Assert.assertEquals(PRICE, results.getPrice());
+  }
+
+  @Test
+  public void analyzeImageAt_mixedCaseTotal_returnsPriceAfterLastTotal()
+      throws IOException, ReceiptAnalysisException {
+    String rawTextWithPriceMixedCaseTotal =
+        "the sub total is $12.23 and the tOtAL is $" + PRICE_VALUE;
+    stubAnnotationResponse(LOGO_CONFIDENCE, rawTextWithPriceMixedCaseTotal);
+    stubTextClassification();
+
+    AnalysisResults results = ReceiptAnalysis.analyzeImageAt(url);
+
+    Assert.assertEquals(PRICE, results.getPrice());
+  }
+
+  @Test
+  public void analyzeImageAt_priceAfterTotalIsNotLargest_returnsPriceAfterTotal()
```

Comment:
Is this named correctly? I'm not understanding this test case.

Reply:
I think I understand it now, maybe it could use a clearer name? Not sure if it would be confusing to others as well or if it's just me!

Reply:
I'm not sure how to make this clearer. Do you have any suggestions?